driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -382,63 +392,102 @@ public CreateCollectionOperation clusteredIndexName(@Nullable final String clust
+            if (t != null) {
+                errHandlingCallback.onResult(null, t);
+            } else {
+                validateCollation(connection, collation, (connection1, t1) ->
+                    new ProcessCommandsCallback(binding, connection1, releasingCallback(errHandlingCallback, connection))
+                            .onResult(null, t1));
             }
         });
     }
 
-    private BsonDocument getCommand(final ConnectionDescription description) {
+    private String getGranularityAsString(final TimeSeriesGranularity granularity) {
+        switch (granularity) {
+            case SECONDS:
+                return "seconds";
+            case MINUTES:
+                return "minutes";
+            case HOURS:
+                return "hours";
+            default:
+                throw new AssertionError("Unexpected granularity " + granularity);
+        }
+    }
+
+    /**
+     * With FLE2 creating a collection can involve more logic and commands.
+     *
+     * <p>
+     *   If the collection namespace has an associated encryptedFields, then do the following operations.
+     *   If any of the following operations error, the remaining operations are not attempted:
+     *   <ol>
+     * <li>Create the collection with name encryptedFields["escCollection"] using default options.
+     *   If encryptedFields["escCollection"] is not set, use the collection name enxcol_.<collectionName>.esc.
+     *   Creating this collection MUST NOT check if the collection namespace is in the AutoEncryptionOpts.encryptedFieldsMap.
+     * <li>Create the collection with name encryptedFields["eccCollection"] using default options.
+     *   If encryptedFields["eccCollection"] is not set, use the collection name enxcol_.<collectionName>.ecc.
+     *   Creating this collection MUST NOT check if the collection namespace is in the AutoEncryptionOpts.encryptedFieldsMap.
+     * <li>Create the collection with name encryptedFields["ecocCollection"] using default options.
+     *   If encryptedFields["ecocCollection"] is not set, use the collection name enxcol_.<collectionName>.ecoc.
+     *   Creating this collection MUST NOT check if the collection namespace is in the AutoEncryptionOpts.encryptedFieldsMap.
+     * <li>Create the collection collectionName with collectionOptions and the option encryptedFields set to the encryptedFields.
+     * <li>Create the the index {"__safeContent__": 1} on collection collectionName.
+     *  </ol>
+     * </p>
+     * @return the list of commands to run to create the collection
+     */
+    private List<Function<ConnectionDescription, BsonDocument>> getCommandFunctions() {
+        if (encryptedFields == null) {
+            return singletonList(this::getCreateCollectionCommand);
+        }
+        return asList(
+                connectionDescription -> new BsonDocument("create", getEncryptedFieldsCollection("esc")),
```

Comment:
https://github.com/mongodb/specifications/pull/1215/ makes this clustered collection.

Reply:
Ack will do in [JAVA-4609](https://jira.mongodb.org/browse/JAVA-4609) - currently blocked until merge.